datastore/writer/core/writer_service.py

```diff
@@ -118,3 +118,29 @@ def truncate_db(self) -> None:
         with self.database.get_context():
             self.database.truncate_db()
             logger.info("Database truncated")
+
+    @retry_on_db_failure
+    def write_action_worker(
+        self,
+        write_request: WriteRequest,
+    ) -> None:
+        """Writes or updates an action_worker-object"""
+        self.write_requests = [write_request]
+
+        with make_span("write action worker"):
+            self.position_to_modified_models = {}
+            with self.database.get_context():
+                position, modified_models = self.write_with_database_context(
+                    write_request
+                )
+                self.position_to_modified_models[position] = modified_models
+
+            with make_span("push events onto redis messaging-bus"):
+                # Only propagate updates to redis after the transaction has finished
+                self.messaging.handle_events(
+                    self.position_to_modified_models,
+                    log_all_modified_fields=False,
+                )
```

Comment:
This is the same code as for the `write` method, right? Then please extract it into its own method

Reply:
No, the `write` method accept write_requests, the `write_action_worker` explicitly 1 write_request.

Reply:
No, this whole part:
```
            self.position_to_modified_models = {}
                with self.database.get_context():
                    for write_request in self.write_requests:
                        position, modified_models = self.write_with_database_context(
                            write_request
                        )
                        self.position_to_modified_models[position] = modified_models

                # Only propagate updates to redis after the transaction has finished
                self.propagate_updates_to_redis(log_all_modified_fields)
```
is the same if I'm not mistaken